### PR TITLE
Oils conflicts

### PR DESCRIPTION
**Pull request by lishaduck**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

It passes brew audit --strict, but I didn't install them as they, well, conflict, so I haven't checked that box.

Related: #173461

### CODE REVIEW

File: Formula/o/oil.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Oil < Formula
 
   depends_on "readline"
 
+  conflicts_with "oils-for-unix", because: "it's the modern version of oils"
```
Comment (SMillerDev) [inline Formula/o/oil.rb:24]: Do they actually conflict? If so, how?

Comment (lishaduck) [reply Formula/o/oil.rb:24]: They both install osh and ysh binaries.

Comment (SMillerDev) [reply Formula/o/oil.rb:24]: Could you make that the reason then?

**Review by branchv — APPROVED** [review]
thanks!

**Pull request merged**